### PR TITLE
chore(weave): Add new` CONTRIBUTING.md`

### PR DESCRIPTION
**Pull request by andrewtruong**
Adds:
- CONTRIBUTING.md

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+| style    | 💅 Style         | Changes that do not affect the meaning of the code (e.g. linting)                   | Maybe        |
+| chore    | ⚙️ Chores        | Changes that do not modify source code (e.g. CI configuration files, build scripts) | No           |
+| revert   | ♻️ Reverts       | Reverts a previous commit                                                           | Maybe        |
+| security | 🔒 Security      | Security fix/feature                                                                | Maybe        |
+
+## Setting up your environment
+
+We use [uv](https://astral.sh/blog/uv) for package and env management. Follow the [uv guide to bootstrap an environment](https://docs.astral.sh/uv/getting-started/installation/), run:
+
+```sh
```
Comment (tssweeney) [inline CONTRIBUTING.md:87]: duplication

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+Then to run a specific environment:
+
+```sh
+nox -e "$TARGET_ENV"  # e.g. nox -e "tests-3.12(shard='trace')"
+```
+
+Tests are split up into shards, which include:
+
+1. `trace` -- all of the trace SDK tests
+2. `trace_server` -- tests for the clickhouse trace server backend
```
Comment (tssweeney) [inline CONTRIBUTING.md:155]: not just clickhouse

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+3. various integrations, like `openai`, `instructor`, etc -- these envs are isolated to simplify testing
+
+### Adding settings
```
Comment (tssweeney) [inline CONTRIBUTING.md:158]: seems out of place here

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+Settings are currently handled on the `UserSettings` object in `weave.trace.settings` and via env vars.
+
+To add a new setting:
+
+1. Add a new field to the `UserSettings` class and provide a short description; and
+2. Add a `should_{new_setting_name}` func to the `settings` module so users can check if that setting is enabled
+
+By default, the env var will be `WEAVE_{setting_name_in_all_caps}`, e.g. `WEAVE_PRINT_CALL_LINK`.
+
+### Adding URLs
```
Comment (tssweeney) [inline CONTRIBUTING.md:169]: seems out of place here

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+To keep URLs consistent, make sure to add and access URLs from `weave.trace.urls`. We currently use functions to return URLs.
+
+### Deprecating features
```
Comment (tssweeney) [inline CONTRIBUTING.md:173]: Does this work? I have never tried it.

File: CONTRIBUTING.md (hunk 2 above)
Comment (andrewtruong) [reply CONTRIBUTING.md:155]: fixed

File: CONTRIBUTING.md (hunk 3 above)
Comment (andrewtruong) [reply CONTRIBUTING.md:158]: ok, removed

File: CONTRIBUTING.md (hunk 4 above)
Comment (andrewtruong) [reply CONTRIBUTING.md:169]: ok, removed

File: CONTRIBUTING.md (hunk 5 above)
Comment (andrewtruong) [reply CONTRIBUTING.md:173]: yeah we use it in a few places, like here:
https://github.com/wandb/weave/blame/andrew/docs4/weave/trace/weave_client.py#L590

**Review by tssweeney — APPROVED** [review]

**Pull request merged**